src/util/hass-translation.ts

```diff
@@ -88,19 +88,18 @@ export function getLocalLanguage() {
       if (language) {
         return language;
       }
+      if (locale.includes("-")) {
+        language = findAvailableLanguage(locale.split("-")[0]);
+        if (language) {
+          return language;
+        }
+      }
     }
   }
   language = findAvailableLanguage(navigator.language);
   if (language) {
     return language;
   }
-  if (navigator.language && navigator.language.includes("-")) {
-    language = findAvailableLanguage(navigator.language.split("-")[0]);
-    if (language) {
-      return language;
-    }
-  }
```

Comment:
I think adding it to `navigator.languages` makes sense, but we should not remove it here? We should instead make a helper function for:

```ts
      language = findAvailableLanguage(locale);
      if (language) {
        return language;
      }
      if (locale.includes("-")) {
        language = findAvailableLanguage(locale.split("-")[0]);
        if (language) {
          return language;
        }
      }
```

Reply:
OK, I see.  I misunderstood what navigator.language was.  I will make a method and call it in both places.

Reply:
I made a change that is a little different.  I couldn't think of a good name for the helper function, and it seemed like something we would want to always check when searching for an available translation, so I added it to that function.

I'll understand if you prefer not to have the recursive call there and want me to break it out differently.  

This is my first time doing a CR on github, first time submitting to HA (or any OS project, for that matter), and first time working with TS, so I'm not super familiar with the industry best practices, nor with the project best practices.  I am more than happy to make changes to make it 

I am in the discord (@Zondebok) if you want to discuss directly.